dashboard/app/models/resource.rb

```diff
@@ -51,4 +58,20 @@ def summarize_for_lesson_plan
       type: type
     }
   end
+
+  private
+
+  def generate_key_from_name
+    key_prefix = name.gsub(/[^a-z0-9\-\_\.]+/, '_')
+    potential_clashes = course_version_id ? Resource.where(course_version_id: course_version_id) : Resource.all
+    potential_clashes = potential_clashes.where("resources.key like '#{key_prefix}%'").pluck(:key)
```

Comment:
in order for this to be an indexed query, we will need to include the course_version_id. I think this is fine to do later, once we are always only storing / looking within a course version, but wanted to call it out now while I'm thinking about it.

Reply:
That makes sense! I have a [ticket](https://codedotorg.atlassian.net/browse/PLAT-434) to fix the course version/resource relationships and will keep this in mind then